src/auth/signout/sign-out.ts

```diff
@@ -0,0 +1,110 @@
+  // TODO: invalidate runtime cookies https://broadworkbench.atlassian.net/browse/IA-3498
+  // sendSignOutMetrics should _not_ be awaited. It's fire-and-forget, and we don't want to block the user's signout
+  sendSignOutMetrics(cause);
+  if (cause === 'expiredRefreshToken' || cause === 'errorRefreshingAuthToken') {
+    notify('info', sessionTimedOutErrorMessage, sessionTimeoutProps);
+  }
+
+  try {
+    const userManager = oidcStore.get().userManager;
+    const redirectUrl = `${Nav.getWindowOrigin()}/${Nav.getLink(signOutCallbackLinkName)}`;
+    // This will redirect to the logout callback page, which calls `userSignedOut` and then redirects to the homepage.
+    userManager!.signoutRedirect({ post_logout_redirect_uri: redirectUrl });
+  } catch (e: unknown) {
+    console.error('Signing out with B2C failed. Falling back on local signout', e);
+    userSignedOut(true);
+    Nav.goToPath('root');
+  }
+};
+
+const sendSignOutMetrics = async (cause: SignOutCause): Promise<void> => {
+  const eventToFire: MetricsEventName = switchCase<SignOutCause, MetricsEventName>(
+    cause,
+    ['requested', () => Events.user.signOut.requested],
+    ['disabled', () => Events.user.signOut.disabled],
+    ['declinedTos', () => Events.user.signOut.declinedTos],
+    ['expiredRefreshToken', () => Events.user.signOut.expiredRefreshToken],
+    ['errorRefreshingAuthToken', () => Events.user.signOut.errorRefreshingAuthToken],
+    ['idleStatusMonitor', () => Events.user.signOut.idleStatusMonitor],
+    ['unspecified', () => Events.user.signOut.unspecified],
+    [DEFAULT, () => Events.user.signOut.unspecified]
+  );
+  const sessionEndTime: number = Date.now();
+  const metricStoreState: MetricState = metricStore.get();
+  const tokenMetadata: TokenMetadata = metricStoreState.authTokenMetadata;
+
+  await Ajax().Metrics.captureEvent(eventToFire, {
+    sessionEndTime: Utils.makeCompleteDate(sessionEndTime),
+    sessionDurationInSeconds:
+      metricStoreState.sessionStartTime < 0 ? undefined : (sessionEndTime - metricStoreState.sessionStartTime) / 1000.0,
+    authTokenCreatedAt: getTimestampMetricLabel(tokenMetadata.createdAt),
+    authTokenExpiresAt: getTimestampMetricLabel(tokenMetadata.expiresAt),
+    totalAuthTokensUsedThisSession: metricStoreState.authTokenMetadata.totalTokensUsedThisSession,
+    totalAuthTokenLoadAttemptsThisSession: metricStoreState.authTokenMetadata.totalTokenLoadAttemptsThisSession,
+  });
+};
+
+export const userSignedOut = (redirectFailed = false) => {
+  // At this point, we are guaranteed to not have a valid token, if the redirect succeeded.
+  cookieReadyStore.reset();
+  azureCookieReadyStore.reset();
+  getSessionStorage().clear();
+
+  if (redirectFailed) {
+    removeUserFromLocalState();
+  }
+
+  const { cookiesAccepted } = authStore.get();
+
+  authStore.reset();
+  authStore.update((state) => ({
+    ...state,
+    signInStatus: 'signedOut',
+    // TODO: If allowed, this should be moved to the cookie store
```

Comment:
Should we create a ticket for this?

Reply:
Ticket created! https://broadworkbench.atlassian.net/browse/ID-1172